truss/server/control/truss_patch/model_code_patch_applier.py

```diff
@@ -0,0 +1,45 @@
+import logging
+import os
+from pathlib import Path
+
+from truss.server.control.patch.types import Action, ModelCodePatch
+
+
+def apply_code_patch(
+    relative_dir: Path,
+    patch: ModelCodePatch,
+    logger: logging.Logger,
+):
+    logger.debug(f"Applying code patch {patch.to_dict()}")
+    filepath: Path = relative_dir / patch.path
+    action = patch.action
+
+    if action in [Action.ADD, Action.UPDATE]:
+        filepath.parent.mkdir(parents=True, exist_ok=True)
+        action_log = "Adding" if action == Action.ADD else "Updating"
+        logger.info(f"{action_log} file {filepath}")
```

Comment:
Suggestion: `f"{action.value()}` (or implicit string conversion of enum member makes sense too...).